data-visualization/pages/03_Community_evolution.py

```diff
@@ -82,7 +88,106 @@ def add_comparisons_evolution():
         st.plotly_chart(fig)
 
 
+def add_comparisons_evolution_grouped_by_contributors_age():
+    """
+    Display the number of active contributors per week, grouped by age.
+    """
+
+    st.markdown("#### Active contributors grouped by age")
+    st.info(
+        "An active contributor is a user who makes at least one public comparison during a given"
+        " period of time.",
+    )
+
+    df = st.session_state.df
+
+    df = df.drop_duplicates(subset=["public_username", "week_date"])[
+        ["public_username", "week_date"]
+    ].reset_index(
+        drop=True
+    )  # Keep only the required data, remove duplicates.
+
+    df.week_date = pd.to_datetime(df.week_date, infer_datetime_format=True, utc=True).astype(
```

Comment:
Using `astype` raises a warning in the console, and will raise an exception in the future.

> Using .astype to convert from timezone-aware dtype to timezone-naive dtype is deprecated and will raise in a future version.  Use obj.tz_localize(None) or obj.tz_convert('UTC').tz_localize(None) instead

I don't know the best practice here (I have nearly zero experience with pandas), but it looks like the `astype` is used to make methods such as `df.week_date.min()` and `.max()` work. It should be possible to keep the timezone aware datetime, right? Maybe by explicitly sorting the column `week_date`?

If there is no side effect we may want to explicitly convert the timezone aware datetime to naive dates. 

In case it helps:

```python
# works
df.week_date = pd.to_datetime(df.week_date, infer_datetime_format=True, utc=True)
df.week_date.cat.as_ordered().max()

# doesn't work
df.week_date = pd.to_datetime(df.week_date, infer_datetime_format=True, utc=True)
df.week_date.max()
*** TypeError: Categorical is not ordered for operation max
you can use .as_ordered() to change the Categorical to an ordered one
```